src/Parser/Shaman/Shared/Talents/StaticCharge.js

```diff
@@ -0,0 +1,35 @@
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'Parser/Core/Analyzer';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+// Static Charge can only reduce the cooldown down to 40 seconds
+const minimumCooldown = 40000;
+const stunReduction = 5000;
+
+class StaticCharge extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+    combatants: Combatants,
+  };
+
+  on_initialized() {
+    this.hasStaticCharge = this.combatants.selected.hasTalent(SPELLS.STATIC_CHARGE_TALENT.id);
```

Comment:
You can set `this.active` here instead of `this.hasStaticCharge`, that way you don't need to check for the talent in the applydebuff below.